src/main/java/net/fabricmc/loader/metadata/CustomValueImpl.java

```diff
@@ -224,12 +369,49 @@ public BooleanImpl(boolean value) {
 		public CvType getType() {
 			return CvType.BOOLEAN;
 		}
+
+		@Override
+		public int hashCode() {
+			return Boolean.hashCode(this.value);
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (this == o) {
+				return true;
+			} else if (!(o instanceof BooleanImpl)) {
+				return false;
+			}
+
+			BooleanImpl aBoolean = (BooleanImpl) o;
+			return this.value == aBoolean.value;
+		}
+
+		@Override
+		public String toString() {
+			return String.valueOf(this.value);
+		}
 	}
 
 	private static final class NullImpl extends CustomValueImpl {
 		@Override
 		public CvType getType() {
 			return CvType.NULL;
 		}
+
+		@Override
+		public boolean equals(Object obj) {
+			return this == NULL;
```

Comment:
This doesn't seem right ;)

Reply:
but its a singleton so it shouldn't matter